app/services/submission/plant_trial_zip_exporter.rb

```diff
@@ -1,12 +1,16 @@
 require 'zip'
 
 class Submission::PlantTrialZipExporter
-  def call(plant_trial)
-    exporter = Submission::PlantTrialExporter.new(
-      OpenStruct.new(submitted_object: plant_trial, user: plant_trial.user)
-    )
+  def call(plant_trial, cache_key)
     compressed_filestream = Zip::OutputStream.write_buffer do |zos|
-      exporter.documents.each do |document_name, content|
+      documents = Rails.cache.fetch(cache_key, expires_in: 300.days) do
```

Comment:
I agree, though caching a stream is not desirable. Hence I opted for merging these two aspects in this single service.